collagraph/renderers/pyside_renderer.py

```diff
@@ -99,11 +99,34 @@ def not_implemented(self, *args, **kwargs):
     raise NotImplementedError(type(self).__name__)
 
 
+class EventFilter(QtCore.QObject):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.event_handlers = defaultdict(set)
+
+    def add_event_handler(self, event, handler):
+        self.event_handlers[event].add(handler)
+
+    def remove_event_handler(self, event, handler):
+        self.event_handlers[event].remove(handler)
+
+    def eventFilter(self, obj, event):
+        event_name = event.type().name.decode()
+        for handler in self.event_handlers[event_name].copy():
+            handler(event)
```

Comment:
I wonder if it would be worthwhile to add:

```suggestion
        if self.event_handlers[event_name]:
            for handler in self.event_handlers[event_name].copy():
                handler(event)
```

Because you're going to be filtering _every event_ on _every qobject_, and those `copy()` calls will add up.

Reply:
Ah, yes, that might be worthwhile, however small the overhead is of copying an empty set. This method will probably indeed be triggered a lot!